### PR TITLE
Catalog of my Things App

### PR DESCRIPTION
**Pull request by krisbobo**
# Milestone: First working version of the project
*The following changes are being introduced:*
- Used the UML class diagram provided to create the required classes and associations between classes in separate files.
- Followed the diagram to create the methods within the respective classes.
- Added unit tests for all implemented methods
- Created a `main.rb` file to serve as the entry point of our console application
- At the start, the console app does the following:
    - Presents the user with a list of options to perform.
    - Lets the user choose an option.
    - If needed, ask for parameters for the option.
- The app presents an option to quit.
- The options presented are as follows:
    - List all books
    - List all music albums
    - List all movies
    - List of games
    - List all genres 
    - List all labels 
    - List all authors    
    - Add a book
    - Add a music album
    - Add a game
- All data from the book, music album, and game is preserved in files using `.json` format.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  last_name VARCHAR(50) NOT NULL,
+  author_id INT,
+  FOREIGN KEY (auothor_id) REFERENCES item(id) ON DELETE CASCADE
+);
+
+-- Create genres table
+CREATE TABLE genres (
+  id INT Generated ALWAYS AS IDENTITY PRIMARY KEY,
+  name VARCHAR(100),
+);
```
Comment (Shedrack-Sunday) [inline schema.sql:44]: - [ ] Good job with the schema and your entire code base in general 👏 .  However, I noticed some of your tables were not exactly implemented as it is on the UML diagram. you're missing the `item:array` properties. Please check the image below and effect changes to the affected tables 👍 

## View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @krisbobo , @AbdusSattar-70 , and @david30-maker       👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  
- Nice use of SQL queries ✔️
- Commendable use of Ruby syntax ✔️
- Clean code structure ✔️   

# Required Changes  🌵 

- [ ] You made some efforts in the implementations which I must say is a job well done. However, there are some issues  I suggest   you make In line with the requirements  

- [ ] Kindly make sure your project video does not exceed `5` mins as specified in the project requirements. You did great, but let's stick to the requirements 👍 
### Instructions:
image


_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: schema.sql (hunk 1 above)
Comment (AbdusSattar-70) [reply schema.sql:44]: Hello @Shedrack-Sunday thanks for your valuable suggestion please give me an example of 
how can we implement it on schema.sql file

Comment (Shedrack-Sunday) [reply schema.sql:44]: Hello @AbdusSattar-70 try this:
``` 
CREATE TABLE Genre (
  id INT,
  name VARCHAR(255),
  items TEXT[]
);
``` 

You can also make it a json

Comment (AbdusSattar-70) [reply schema.sql]: @Shedrack-Sunday Thanks for your example suggestion  👍

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Shedrack-Sunday) [reply schema.sql]: You're welcome

**Pull request merged**